### ISSUE TITLE
Can we use tex in labels / titles ?

### ISSUE THREAD

**Issue by BowenFu** [racket/plot#58]
Something like
(plot (function sin (- pi) pi #:label "y = \\sin(x^2)"))?

![untitled](https://user-images.githubusercontent.com/5812640/74582019-5a5d5480-4ff1-11ea-8995-7deff69d1ca5.png)

**Comment by soegaard** [comment]
Den lør. 15. feb. 2020 kl. 05.48 skrev Bowen Fu <notifications@github.com>:

> Something like
> (plot (function sin (- pi) pi #:label "y = \sin(x^2)"))?
>
> [image: untitled]
> <https://user-images.githubusercontent.com/5812640/74582019-5a5d5480-4ff1-11ea-8995-7deff69d1ca5.png>
>
I offer one piece of the puzzle, namely a function that turns a string
containing latex formula
into a `pict`:

https://github.com/soegaard/latex-pict

It relies on `racket-poppler` which contains bindings for Poppler (a
library for
drawing pdfs using Cairo - which is the graphics engine Racket uses).

Note that `plot` offers `plot/dc` which allows you to draw the plot on a
drawing context `dc`.
This makes it possible to combine picts and plots. For approximate
placement of labels,
I think this is enough.

If you need to place labels using plot coordinates, then some work needs to
be done:
a function to convert dc coordinates to plot coordinates (and vice versa)
needs to be written.

The tricky bit is that plot (when given a width and height) uses some of
the same
for margins and labels. How much space is determined here(ish):

https://github.com/racket/plot/blob/9823d1ab9ff9144a8a3114afc0af6a69a0fbc4b7/plot-lib/plot/private/plot2d/plot-area.rkt#L533

I think, this needs to be added to `plot` itself (correct me if I am wrong).

/Jens Axel

**Comment by alex-hhh** [comment]
You can use unicode characters for some math notation.  this is not as flexible as using TeX, but can be useful in a lot of cases:

```racket
#lang racket
(require plot)
(plot (function sin (- pi) pi #:label "y = sin(x²)"))
```

Produces:

<img width="333" alt="i58" src="https://user-images.githubusercontent.com/11592690/85803550-f6565400-b779-11ea-89c3-7b53829efb20.png">

**Comment by alex-hhh** [comment]
As for the implementation, the `#label` argument to the plot functions can be extended to accept a `pict` argument in addition to a string.  There is already a [point-pict](https://docs.racket-lang.org/plot/renderer2d.html?q=point-pict#%28def._%28%28lib._plot%2Fmain..rkt%29._point-pict%29%29) function which can display `pict` objects on a plot.

This way, the latex-pict package suggested above could be used to render the labels.

**Comment by alex-hhh** [comment]
The plot package has been extended to support pictures for legend entries, names for axes and plot titles.  This allows formatting these elements using pretty much anything that can be rendered.  In particular it would allow the use of `latex-pict` to render LaTeX formulas.

As I noted previously, you can also use Unicode for the strings, which allows some limited mathematical notation.  E.g. you can use `x²` directly.

This functionality will be available in the next Racket version (7.9).
